jena-arq/src/main/java/org/apache/jena/sparql/expr/NodeValue.java

```diff
@@ -954,7 +971,8 @@ public boolean isDayTimeDuration()
     public boolean     getBoolean()     { raise(new ExprEvalTypeException("Not a boolean: "+this)) ; return false ; }
     public String      getString()      { raise(new ExprEvalTypeException("Not a string: "+this)) ; return null ; }
     public String      getLang()        { raise(new ExprEvalTypeException("Not a string: "+this)) ; return null ; }
-    
+    public String      getCollation()   { raise(new ExprEvalTypeException("Not a collation: "+this)) ; return null ; }
```

Comment:
Nice catch! I'm updating the code, experimenting with the idea of having a `NodeValueSortKey` type instead. If that works well, will update the message to "Not a sort key"...